src/utils.py

```diff
@@ -1,15 +1,54 @@
 import logging
-from typing import Sequence, Any
+from typing import Any, Optional, Sequence, Tuple, Union, cast
 from io import BytesIO
 from pathlib import Path
 import numpy as np
 
-
+from cloudpathlib import S3Path
+from tqdm.auto import tqdm
 from cpr_data_access.parser_models import BlockType, ParserOutput, TextBlock
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def build_indexer_input_path(indexer_input_dir: str, s3: bool) -> Union[S3Path, Path]:
+    _LOGGER.info(f"Tasks will be retrieved from {'s3' if s3 else 'local'}: {indexer_input_dir}")
+    if s3:
+        indexer_input_path = cast(S3Path, S3Path(indexer_input_dir))
+    else:
+        indexer_input_path = Path(indexer_input_dir)
+    return indexer_input_path
+
+
+def get_index_paths(
+    indexer_input_path: str,
+    files_to_index: Optional[str] = None,
+    limit: Optional[int] = None,
+) -> Tuple[Sequence[ParserOutput]]:
+
+    files_to_index = files_to_index.split(",") if files_to_index else []
+
+    paths = []
+    doc_ids = []
+    for i, path in enumerate(tqdm(list(indexer_input_path.glob("*.json"))), 1):        
```

Comment:
Just a flag. I find `tqdm` is great when running locally - but a complete pain in the A** when reading cloud logs :grin:

Reply:
Yep good point, it probably makes even less sense now that I've moved the flow around as well. It also feels like its adding some noise to the code that doesn't need to be there. I'm happy to get rid of it.